themes/edx.org/lms/templates/dashboard.html

```diff
@@ -146,7 +147,7 @@
       <h2 class="header-courses">${_("My Courses")}</h2>
     </header>
     % if len(course_entitlements + course_enrollments) > 0:
-        <ul class="listing-courses" data-programs=${programs_data}>
+        <ul class="listing-courses" data-programs='${json.dumps(programs_data)}'>
```

Comment:
I thought there were two places to make this change?

Reply:
i initially changed that but it caused some test errors for some reason, so I changed it back, maybe something is different due to the way it is being used here?
    <%static:require_module module_name="js/views/message_banner" class_name="MessageBannerView">
        var banner = new MessageBannerView({urgency: 'low', type: 'warning'});
        $('#content').prepend(banner.$el);
        banner.showMessage(${redirect_message | n, dump_js_escaped_json})
    </%static:require_module>

Reply:
The tag `<%static:require_module>` sets up a JavaScript context.  For example, the line `banner.showMessage(...)` is JavaScript.  So, this needs to be JavaScript-escaped.

This is different from a data attribute which takes place in an HTML context and needs to be HTML-escaped, regardless of whether or not you happen to be storing JSON in the attribute.

It is more about how the browser processes each part of the page.

Thanks.